pages/publicfiles/PublicFilesHandler.inc.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+/**
+ * @file pages/publicfiles/PublicFilesHandler.inc.php
+ *
+ * Copyright (c) 2013-2015 Simon Fraser University Library
+ * Copyright (c) 2003-2015 John Willinsky
+ * Distributed under the GNU GPL v2. For full terms see the file docs/COPYING.
+ *
+ * @class PublicFilesHandler
+ * @ingroup pages_publicfiles
+ *
+ * @brief Handle requests for public files functions.
+ */
+
+import('classes.handler.Handler');
+import('lib.pkp.classes.file.FileManager');
+
+class PublicFilesHandler extends Handler {
+	
+	/* @var fileManager FileManager object (see lib/pkp/classes/file/FileManager.inc.php) */
+	var $fileManager;
+	
+	//Constructor
+	function PublicFilesHandler() {
+		$this->fileManager = new FileManager();
+	}
+	
+	/**
+	 * Delete a file in a user's public files directory
+	 * @param $args array
+	 * @param $request PKPRequest
+	 * @return bool
+	 */
+	function delete($args, $request) {
```

Comment:
If I am not overlooking this, it will allow for any logged in user to delete any file inside the public file folder of a journal. I think that's a flaw.